### PR TITLE
convert itol functionality into qiime2 command

### PR DESCRIPTION
**Pull request by qiyunzhu**
Hi @anupriyatripathi I have created the first PR to address issue #100 . It works. It generates a feature table which can be drag & drop into iTOL interface and render as a multi-value bar chart.

However, I refrained from adding the remaining functionality: tip label and clade color, into this module, because I am not sure how to handle it right. As you know, QIIME 2's output file is always .qza. However a .qza file cannot be parsed by iTOL (except for feature table). One still need to unzip it before use. Let alone the trouble of creating two more Artifact types.

I think I may not fully understand the philosophy of QIIME 2 plugin design. I would like to know your opinion of how to proceed. Thank you!

### CODE REVIEW

**Comment by anupriyatripathi** [comment]
@qiyunzhu thanks for starting the PR. I think we might be able to use the semantic type FeatureData[Molecules] for the two other files. Let's chat in person this week to brainstorm how to best engineer this!

**Comment by qiyunzhu** [comment]
@anupriyatripathi To add unit test I will need some feature table and metadata. Do you happen to have them? (needs to be small to fit into data/ directory).

File: q2_qemistree/_itol.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        Categorical sample metadata column.
+
+    Returns
+    -------
+    biom.Table
+        Table of mean feature frequencies per category per sample.
+    '''
+    column = metadata.drop_missing_values()
+    catmap = column.to_series().to_dict()
+    return table.collapse(lambda i, _: catmap[i], norm=True, axis='sample')
```
Comment (ElDeveloper) [inline q2_qemistree/_itol.py]: Just in case this is useful, `qiime feature-table group` can collapse samples by a metadata category.

**Comment by anupriyatripathi** [comment]
@qiyunzhu yes, I'm sending you a small feature table and metadata for unit-testing. Thanks!

**Pull request closed without merging**